### PR TITLE
Consider both ij and ji pairs in RDF calculations

### PR DESCRIPTION
**Pull request by clpetix**

### CODE REVIEW

**Comment by clpetix** [comment]
@mphoward this should have everything ported and it seems to be working in both HOOMD and LAMMPS!

File: src/relentless/simulate/lammps.py [hunk 1]
```diff
@@ -1164,22 +1164,50 @@ def process(self, sim, sim_op):
...
+                        # neighbors of j that are type i
+                        if j != i:
+                            filter_ji = numpy.logical_and(
+                                type_masks[j][neighbors[:, 0]],
+                                type_masks[i][neighbors[:, 1]],
+                            )
+                            filter_ij = numpy.logical_or(filter_ij, filter_ji)
+                        neighbors_ij.filter(filter_ij)
+                        print("types: ", i, j)
+                        print(neighbors_ij[:])
```
Comment (mphoward) [inline src/relentless/simulate/lammps.py:1202]: Please remove these debug statements!

**Review by mphoward — APPROVED** [review]
Thanks! This all looks good to me after you remove the debug prints. Did you run this on a WCA mixture to make sure everything looks good?

**Comment by clpetix** [comment]
@mphoward I have a working version of the code ported to both LAMMPS and HOOMD and it replicates the results from before! 
image

**Comment by clpetix** [comment]
@mphoward this is ready for a review whenever you get a chance!

**Comment by clpetix** [comment]
> Looks great! Just a few small things to clean up and then we can merge. Just to confirm: did you run a 2d WCA simulation in addition to the 3d check?

Yes, I tried all three cases with both engines in both 2d and 3d boxes and they are in agreement! Here are the 2-d cases: 

image

**Comment by clpetix** [comment]
@mphoward the changes have been applied and this should be ready to go!

**Review by mphoward — APPROVED** [review]
Looks great, thanks! Can you merge `main` into the box resize branch for Beth after I merge this PR? We should also do this for Tyla's, but that is going to cause merge conflicts that we will need to resolve so I would rather punt on that for now.

**Pull request merged**